packages/umpire/package.py

```diff
@@ -237,6 +244,7 @@ class Umpire(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on("camp+openmp", when="+openmp")
     depends_on("camp~cuda", when="~cuda")
     depends_on("camp@main", when="@develop")
+    depends_on("camp@2024.07.0:", when="@2024.07.0:")
     depends_on("camp@2024.02.0:", when="@2024.02.0:")
```

Comment:
Are the unbounded `when` parameters still right?

Specifically for umpire,raja,camp when raja is `2024.02.0` and above?

Reply:
Not sure. I thought, the version constraints would be parsed in order that they appear. What is the best practice?

Reply:
I’m gonna ask it differently:
Would it be too restrictive to enforce coherency between the versions by default ?
Camp, Umpire, RAJA and CHAI being release jointly, they are tested to work with one another when the version is the same. It seems _safer_ to enforce that with something like
```
depends_on("camp@2024.07.0:", when="@2024.07.0:")
depends_on("camp@2024.02.0:camp@2024.02.99", when="@2024.02.0:2024.02.99")
```
Being safer doesn’t make it right if in production we happen to mix versions for good reasons...

Reply:
For the records, when no version is specified, Spack will try to use the latest release, it does not depend on the ordering.
The scenario of interest here is:
"If a spec asks for `raja@2024.02.0` is it right for Spack to pick `chai@2024.07.0` (or similarly with camp and raja), which is what would happen with the current implementation."

Reply:
Here for example, `raja@2024.02.0 ^camp@2024.07.0` is not forbidden. (same for Umpire)

Reply:
@rhornung67 @davidbeckingsale I cannot make that decision myself. To me two options are possible:
- We prevent only what will fail for sure.
  - Pro: no artificial restrictions on the version choice, users may mix versions.
  - Con: Each release of any project should make it clear whether there is a breaking change or not. E.g., there is not indication in camp release logs (https://github.com/LLNL/camp/releases) about why 2024.02 cannot be used in place of 2023.06, while the RAJA package has this constraint:
```python
      depends_on("camp@2024.02.0:", type="build", when="@2024.02.0:")
      depends_on("camp@2023.06.0", type="build", when="@2023.06.0:2023.06.1")
```
- We enforce the use of a coherent set of specs. (Updating any of Camp, Umpire, RAJA,... will result in the update of all of them).
  - Pro: we make certain no exotic combination are used
  - Con: we prevent combinations that actually work.

Reply:
I am working on adding the latest release of the radiuss packages directly in Spack so that the Spack CI can be used to find issues:
https://github.com/spack/spack/pull/45648

In particular, I am seeing a failure with CHAI not finding RAJA:
https://gitlab.spack.io/spack/spack/-/jobs/12157661

Reply:
@adrienbernede when you have a minute, please take a look at this. I'm not 100% clear on what you preference is for version constraints older than the latest release. I followed what was in the packages before I made changes.

Reply:
@adrienbernede I think the failure with CHAI not finding RAJA may be fixed in this PR branch now. CHAI version constraints included a RAJA version tag that does not exist: https://github.com/LLNL/radiuss-spack-configs/pull/107/commits/b91607553d1d88818304c582f997dae541fdf456

Reply:
@rhornung67 I fixed the CHAI issue in the spack PR mentioned above, and reported that fix in RADIUSS Spack Configs through https://github.com/LLNL/radiuss-spack-configs/pull/109